src/particles.cpp

```diff
@@ -185,11 +185,16 @@ void Particle::updateLight()
 	else
 		light = blend_light(m_env->getDayNightRatio(), LIGHT_SUN, 0);
 
-	u8 m_light = decode_light(light + m_glow);
-	m_color.set(255,
-		m_light * m_base_color.getRed() / 255,
-		m_light * m_base_color.getGreen() / 255,
-		m_light * m_base_color.getBlue() / 255);
+	bool fullbright = g_settings->getBool("fullbright_mode");
```

Comment:
This setting getter is too slow to use here, fullbright mode needs to be cached in a bool.